packages/frontend/src/components/merchants/ShopSettings.tsx

```diff
@@ -186,30 +186,39 @@ export default function ShopSettings() {
   }
 
   function handleAcceptedCurrencies(e: ChangeEvent<HTMLInputElement>) {
-    const [addr, chainId] = e.target.value.split("/");
-    const address = addr as Address;
+    const [address, chainId] = e.target.value.split("/");
+    const newAcceptedCurrencies = new Map(acceptedCurrencies);
+    const allChainAddresses = newAcceptedCurrencies.get(chainId) || new Map();
+
     if (e.target.checked) {
-      // Add to accepted currencies
-      setAcceptedCurrencies([
-        ...acceptedCurrencies,
-        { address, chainId: Number(chainId) },
-      ]);
+      allChainAddresses.set(address, null);
+      newAcceptedCurrencies.set(chainId, allChainAddresses);
+      setAcceptedCurrencies(newAcceptedCurrencies);
     } else {
-      // Remove from accepted currencies
-      const test = acceptedCurrencies.filter((c: Currency) => {
-        return !(c.address.toLowerCase() === address.toLowerCase() &&
-          c.chainId === Number(chainId));
-      });
+      if (allChainAddresses) {
+        //TODO: may have to delete address as lowercase
```

Comment:
This might be a good case to overload the map with custom get/set that do the lower-casing.

In general it's fine to lower-case the addresses. The casing is actually "just" a checksum: `https://coincodex.com/article/2078/ethereum-address-checksum-explained/`

Reply:
so both versions (capitalized or binary) are functionally equivalent as far as the value of the address is concerned. The checksum also disappears when going from hex to binary.